core/app/models/spree/variant.rb

```diff
@@ -40,8 +40,9 @@ class Variant < Spree::Base
     after_create :set_master_out_of_stock, unless: :is_master?
 
     after_touch :clear_in_stock_cache
+    after_save :update_counter_cache
 
-    scope :in_stock, -> { joins(:stock_items).where('count_on_hand > ? OR track_inventory = ?', 0, false) }
+    scope :in_stock, -> { joins(:stock_items).where('spree_stock_items.count_on_hand > ? OR track_inventory = ?', 0, false) }
```

Comment:
Line is too long. [125/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.